app/models/auth_user.rb

```diff
@@ -32,18 +32,21 @@ def from_token_payload(payload)
       team = team_from_payload(payload)
       role = role_from_payload(payload)
       email = email_from_payload(payload)
-
-      auth_user = CreateAnonymousUser.run!(
-        auth0_id_string: id,
-        team_name: team,
-        role_title: role,
-        access_level: access_level,
-        email: email
-      )
+      auth_user = nil
+
+      RedisMutex.with_lock("CreateAnonymousUser:#{id}") do
```

Comment:
would it make sense to add this to the .run! itself? As in, we create some sort of decorator that actually mutexes a use_case (like we did for the redis caching).

Reply:
hmm ah well, essentially that's what you're doing here, only a new level of abstraction

Reply:
Yep. I kind of like that it's obvious. It's like the ActiveRecord::Base.Transaction blocks, but better.